src/app-layer-parser.c

```diff
@@ -221,6 +226,9 @@ int AppLayerParserSetup(void)
     SCReturnInt(0);
 }
 
+// default value is 0x1000
+uint32_t g_alparser_tx_max = 4096;
```

Comment:
thinking about this a bit, it seems to make sense to make this a per alproto setting, where some protocols will use much lower defaults